### PR TITLE
spec: update config route response with market variables

### PR DESCRIPTION
**Pull request by chappjc**
Resolves https://github.com/decred/dcrdex/issues/126

Code changes are in PR https://github.com/decred/dcrdex/pull/129

### CODE REVIEW

**Comment by chappjc** [comment]
Oh, this is draft.

**Comment by chappjc** [comment]
This is ready now.

**Comment by chappjc** [comment]
Also needs to have registration fee moved to 'config' from 'register'.

**Comment by chappjc** [comment]
Registration fee added to config response, but what do you think about keeping `fee` in the `register` response too?  It makes sense to have the fee amount included with the provided fee address.

File: spec/fundamentals.mediawiki [hunk 1]
```diff
@@ -7,15 +7,37 @@ peer-to-peer trades with the mechanics of order execution existing entirely on
...
+The cancellation ratio for a user is the ratio of their canceled order count to
+completed order count. A user's cancellation ratio must remain below the configured
+[[community.mediawiki/#Rule_3_An_accounts_cancellation_ratio_must_remain_below_the_threshold|'''cancellation threshold''']]
+(<code>cancelmax</code>) to avoid penalty.
+
+The '''broadcast timeout''' (<code>btimeout</code>) is the amount of time a
+client has to broadcast a transaction. For the maker, the broadcast time is
+measured from the time of match notification. For the taker, timeout is measured
+from the time when the maker's swap receives its
+<code>swapconf</code><sup>th</sup> confirmation (see [[#Asset_Variables]]).
```
Comment (buck54321) [inline spec/fundamentals.mediawiki]: Did you want this link to display with the anchor and underscore?

File: spec/fundamentals.mediawiki [hunk 2]
```diff
@@ -7,15 +7,37 @@ peer-to-peer trades with the mechanics of order execution existing entirely on
...
+
+The user's transaction that pays the registration fee specified by the
+<code>register</code> response during [[accounts.mediawiki/#Account_Creation|Account Creation]]
```
Comment (buck54321) [inline spec/fundamentals.mediawiki]: ~~Account Creation~~ -> account creation

File: spec/fundamentals.mediawiki [hunk 3]
```diff
@@ -155,6 +204,7 @@ The goal of this algorithm is to ensure that an individual client cannot obtain
 a deterministic latency advantage over other clients when executing trades.
 Limiting this possibility mitigates advantages gained from front-running,
 spoofing, and other manipulative trading practices.
+Each market will have an epoch duration set by the operator.
```
Comment (buck54321) [inline spec/fundamentals.mediawiki]: Sentence seems out-of-place here. Perhaps just mention the variable in the `===Market Variables===` section instead.

Comment (chappjc) [reply spec/fundamentals.mediawiki]: Yeah, I agree.  It's already mentoined in Market Variables, so removing this line.

File: spec/fundamentals.mediawiki (hunk 1 above)
Comment (chappjc) [reply spec/fundamentals.mediawiki]: No, that was just sloppy.

File: spec/fundamentals.mediawiki [hunk 4]
```diff
@@ -7,15 +7,37 @@ peer-to-peer trades with the mechanics of order execution existing entirely on
...
+
+The '''broadcast timeout''' (<code>btimeout</code>) is the amount of time a
+client has to broadcast a transaction. For the maker, the broadcast time is
+measured from the time of match notification. For the taker, timeout is measured
+from the time when the maker's swap receives its
+<code>swapconf</code><sup>th</sup> confirmation (see [[#Asset_Variables|Asset Variables]]).
+
+The user's transaction that pays the registration fee specified by the
+<code>register</code> response during [[accounts.mediawiki/#Account_Creation|account creation]]
+must have <code>regfeeconfirms</code> confirmations to be considered paid.
```
Comment (buck54321) [inline spec/fundamentals.mediawiki]: It actually must have `regfeeconfirms` before the server will even accept the `notifyfee`, though the distinction isn't too important here.

File: spec/fundamentals.mediawiki [hunk 5]
```diff
@@ -7,15 +7,37 @@ peer-to-peer trades with the mechanics of order execution existing entirely on
 the separate blockchains via atomic swaps.
 These are:
 * Asset-specific order quantity increments and transaction fee rates
-* Epoch-based pseudorandom order matching
+* Epoch-based pseudorandom order matching, with epoch duration configure per-market
```
Comment (buck54321) [inline spec/fundamentals.mediawiki]: Is **configure** supposed to be **configured**? I'm also not sure that this detail is necessary here.

File: spec/fundamentals.mediawiki [hunk 6]
```diff
@@ -7,15 +7,39 @@ peer-to-peer trades with the mechanics of order execution existing entirely on
...
+client has to broadcast a transaction. For the maker, the broadcast time is
+measured from the time of match notification. For the taker, timeout is measured
+from the time when the maker's swap receives its
+<code>swapconf</code><sup>th</sup> confirmation (see [[#Asset_Variables|Asset Variables]]).
+
+The user's transaction that pays the registration fee specified by the
+<code>register</code> response during [[accounts.mediawiki/#Account_Creation|account creation]]
+must have <code>regfeeconfirms</code> confirmations before the server will
+accept the user's [[accounts.mediawiki/#Step_2_Fee_Notification|fee payment notification]]
+and complete the registration process.
```
Comment (chappjc) [inline spec/fundamentals.mediawiki]: Please see if the wording for this more detailed description of the process is clear. @buck54321

**Pull request merged**